packages/ens-controller/src/EnsController.ts

```diff
@@ -86,6 +118,25 @@ export class EnsController extends BaseControllerV2<
         ...state,
       },
     });
+
+    if (provider && onNetworkStateChange) {
+      onNetworkStateChange((networkState) => {
+        const currentNetwork =
+          networkState.network === 'loading' ? null : networkState.network;
+        if (currentNetwork && this.#getNetworkEnsSupport(currentNetwork)) {
+          const networkish: Networkish = {
+            chainId: convertHexToDecimal(networkState.providerConfig.chainId),
+            name: NETWORK_ID_TO_ETHERS_NETWORK_NAME_MAP[
+              currentNetwork as unknown as keyof typeof NETWORK_ID_TO_ETHERS_NETWORK_NAME_MAP
```

Comment:
Rather than cast here, we can use a type guard to check whether this is a known network ID.

A good place to use that guard would be in the above condition, where we're checking to ensure `currentNetwork` is truthy`.

e.g. instead of `if (currentNetwork && ...)`, it could be `if (isKnownNetworkId(currentNetwork) && ...` (or `isEthersNetworkId` or whatever you want to call it).

Then that type guard could look something like this:

```
isKnownNetworkId(networkId: string): networkId is keyof typeof NETWORK_ID_TO_ETHERS_NETWORK_NAME_MAP {
  return Boolean(NETWORK_ID_TO_ETHERS_NETWORK_NAME_MAP[networkId]);
}
```

The `is` operator in the return type is a type predicate, documented here: https://www.typescriptlang.org/docs/handbook/2/narrowing.html#using-type-predicates
Type predicates can be used in cases where TypeScript's inference isn't effective enough at narrowing the type, like in this case.